### ISSUE TITLE
Incorporate AxCrypt-SHA1 into raw SHA-1 source file.

### ISSUE THREAD

**Issue by magnumripper** [magnumripper/JohnTheRipper#2006]
https://github.com/magnumripper/JohnTheRipper/commit/9c6373b020fe92e4210777962505bee33701ef1b#commitcomment-15601450

01e053b

**Comment by magnumripper** [comment]
That commit doesn't work for me
```
$ ../run/john -test -form:raw-sha1-axcrypt 
Benchmarking: Raw-SHA1-AxCrypt [SHA1 128/128 AVX 4x]... FAILED (source)
```
Probably trivial problem though.

**Comment by claudioandre** [comment]
Have you enabled any of the two tests?
```
	//{"{SHA}VaiuD3vBn9alvHyZcuPY0/UulBo=", "base64test"},
	//{"{SHA}COqO1HN3nVE2Fh45LQs05wAAAAA=", "base64test0truncated"},
``` 

If so, they will work fine if you try to crack them in a real run. I tested this here and on super.
****

This error means self test ciphertext() != source(). I havent fixed it because I'm not sure which is the correct. Basically:
- it cracks them (real run and self test). The problem happens only in the check shown above.
- I don't know what is the canonical form the format should have (there is one?).
- E.g., is this 'wAAAAA=' correct? Is '3dhUj0=' correct? Any value is acceptable (last N bytes are ignored).

**Comment by claudioandre** [comment]
If you have not commented these tests. Please, can you answer?:
- is there a canonical?
- why I haven't seen the error you are seeing?

**Comment by magnumripper** [comment]
I did not look at (or change) the code at all yet, just built and tried to run it as-is. But I think I see the main problem. The self-test requires (off the top of my head) that the output of source() exactly matches the original ciphertext as it looked after split().

I need to look closer to this but in the end it's very trivial and, like you said, mostly about deciding what to do.

**Comment by claudioandre** [comment]
Ok, I decided that the last byte is alwasy zero (in source and self text cipher text).
e91b81ebe143a98e69f90b2db92e67e3a48ffe58

Does it work to you? It does here and on super

**Comment by magnumripper** [comment]
That works. But I suppose it now needs input hashes padded with zeros? That is inconvenient for user. I presume the native format is simply 32 characters of hex.

**Comment by magnumripper** [comment]
OTOH the currently committed format has the same padding..

**Comment by magnumripper** [comment]
I committed it (bleeding dubbed it c5d7b91 but it's the same commit). We'll see what to do with that padding in the end.

**Comment by claudioandre** [comment]
>I suppose it now needs input hashes padded with zeros?

No, it doesn't. You can crack a pure RAW1 hash using AxCrypt. Only self test hashes have to be padded.

**Comment by magnumripper** [comment]
Yes but I meant just 128 bits of input, as in `145595ef8b1d96d7bd9c5ea1c6d28766` as opposed to anything longer. I actually have no idea but I *presume* that's how these hashes look straight from their "source".

**Comment by jfoug** [comment]
a5b6a9b

We probably should 'fail' in valid for hashes that are 40 byte hex, and top 8 bytes are NOT 0.

**Comment by magnumripper** [comment]
Unless zero-padded 40-byte ones are seen in the wild, we shouldn't support them at all. Only 32-byte truncated ones.

**Comment by jfoug** [comment]
The original 'how to' page listed to append 0 bytes to the 32  byte hex.  Who knows what will be seen ITW

**Comment by magnumripper** [comment]
I would say that clearly indicates they come as 32 byte hex.

**Comment by Fist0urs** [comment]
Hi, sorry for the delay

@claudioandre 
> No, it doesn't. You can crack a pure RAW1 hash using AxCrypt. Only self test hashes have to be padded.

you are right, but "original" axcrypt-sha1 should be padded (see below)

@magnumripper 
> I would say that clearly indicates they come as 32 byte hex.

Yes, they are 32 bytes hex hashes, as found in memory, but I had to pad to 40 bytes to fit raw-sha1 default implementation

@jfoug 
> We probably should 'fail' in valid for hashes that are 40 byte hex, and top 8 bytes are NOT 0.

That could be an option yes, but I think it should be better to only check the first 32 hex bytes (eg: you've already cracked a raw-sha1 and this one is present in your .pot, then we should be able to say "ok this one is already cracked" to shorten crack tests)

**Comment by jfoug** [comment]
Could we add linked-in raw-sha1 to the raw-sha1 format, OR would the reverse steps be a problem since we have to check the 2nd 4 byte value ?  I think the split/prepare/valid should be easy.  The source would likely simply copy over from the current format. But it is the hash and cmp_*() functions which I question if we can do easily.
